src/geometry.cpp

```diff
@@ -51,7 +51,9 @@ Geometry::Geometry()
 
 Geometry::~Geometry()
 {
-    destroyBuffers();
+    // Line below commented out as a temporary fix to prevent segfaults 
+    // when removing datasets. Probably leaking memory now instead though.
+    //destroyBuffers();
```

Comment:
Could you put this back since it's a separate issue, and this obviously leaking isn't a viable solution (by all means, keep the hack in your local copy until we can get to the root cause).